src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -747,6 +853,38 @@ private TcpConnectionInfo GetConnectionInfo(ReplicationNode node, bool external)
             return null;
         }
 
+        public X509Certificate2 GetCertificateForReplication(ReplicationNode node, out TcpConnectionHeaderMessage.AuthorizationInfo authorizationInfo)
+        {
+            authorizationInfo = null;
+            var exNode = node as ExternalReplication;
+            if (exNode == null)
+                return _server.Server.Certificate.Certificate;
+
+            if (exNode.PullReplicationAsEdgeSettings?.CertificateThumbprint != null)
+            {
+                using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var key = Constants.Certificates.FeaturePrefix + exNode.PullReplicationAsEdgeSettings.CertificateThumbprint;
+                    var certificate = _server.Cluster.Read(context, key);
+                    if (certificate == null)
+                    {
+                        return _server.Server.Certificate.Certificate;
```

Comment:
Is this correct? When specified key does not exist then we want to use the server certificate? If yes, then probably we should check users intention with `string.IsNullOfWhiteSpace(CertificateThumbprint)` ?